dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -93,25 +96,41 @@ public SqliteVectorStoreRecordCollection(
         this._dataTableName = this.CollectionName;
         this._vectorTableName = GetVectorTableName(this._dataTableName, this._options);
 
-        this._propertyReader = new VectorStoreRecordPropertyReader(typeof(TRecord), this._options.VectorStoreRecordDefinition, new()
-        {
-            RequiresAtLeastOneVector = false,
-            SupportsMultipleKeys = false,
-            SupportsMultipleVectors = true
-        });
-
-        // Validate property types.
-        this._propertyReader.VerifyKeyProperties(SqliteConstants.SupportedKeyTypes);
+        this._model = new VectorStoreRecordModelBuilder(SqliteConstants.ModelBuildingOptions)
+            .Build(typeof(TRecord), this._options.VectorStoreRecordDefinition);
 
-        this._vectorPropertiesExist = this._propertyReader.VectorProperties.Count > 0;
+        this._vectorPropertiesExist = this._model.VectorProperties.Count > 0;
 
-        this._dataTableProperties = new(() => [this._propertyReader.KeyProperty, .. this._propertyReader.DataProperties]);
-        this._vectorTableProperties = new(() => [this._propertyReader.KeyProperty, .. this._propertyReader.VectorProperties]);
+        // Populate some collections of properties
+        this._keyStorageName = this._model.KeyProperty.StorageName;
 
-        this._dataTableStoragePropertyNames = new(() => [this._propertyReader.KeyPropertyStoragePropertyName, .. this._propertyReader.DataPropertyStoragePropertyNames]);
-        this._vectorTableStoragePropertyNames = new(() => [this._propertyReader.KeyPropertyStoragePropertyName, .. this._propertyReader.VectorPropertyStoragePropertyNames]);
+        foreach (var property in this._model.Properties)
+        {
+            switch (property)
+            {
+                case VectorStoreRecordKeyPropertyModel keyProperty:
+                    this._dataTableProperties.Add(keyProperty);
+                    this._vectorTableProperties.Add(keyProperty);
+                    this._dataTableStoragePropertyNames.Add(keyProperty.StorageName);
+                    this._vectorTableStoragePropertyNames.Add(keyProperty.StorageName);
+                    break;
```

Comment:
I assume that in the long term we will be able to remove all these fields and simply use model everywhere?